subprojects/enterprise-operations/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteTaskBuildOperationType.java

```diff
@@ -62,6 +62,16 @@ public interface Result {
         @Nullable
         String getSkipMessage();
 
+        /**
+         * The detailed reason why the task was skipped, if provided by the project configuration.
+         *
+         * Expected to be {@link org.gradle.api.tasks.TaskState#getSkipReasonMessage()}.
+         *
+         * @since 7.6
+         */
+        @Nullable
+        String getSkipReasonMessage();
```

Comment:
Is there anything that we can codify about the presence of this value? e.g. is it only present with #getSkipMessage() has a certain value?

Reply:
Yes, at least for now, the skip reason message is set if and only if the outcome (skip message) is `SKIPPED`. If the task is skipped but the user has not provided a reason message (e.g. `onlyIf { false }`), then there goes a generated message like `'Task satisfies onlyIf spec' not satisfied`.

Reply:
@h0tk3y Can you please include this information in the Javadoc here.